pyvista/core/pointset.py

```diff
@@ -1076,8 +1107,47 @@ def n_faces(self) -> int:  # numpydoc ignore=RT01
         4
 
         """
+        if _USE_STRICT_N_FACES:
+            return self.n_faces_strict
+
+        # Only issue the deprecated n_faces warning the first time it's used
+        global _WARNED_DEPRECATED_NONSTRICT_N_FACES
+        if not _WARNED_DEPRECATED_NONSTRICT_N_FACES:
+            _WARNED_DEPRECATED_NONSTRICT_N_FACES = True
+
+            # deprecated 0.43.0, convert to error in 0.46.0, remove 0.49.0
+            warnings.warn(
+                """The current behavior of `pv.PolyData.n_faces` has been deprecated.
+                Use `pv.PolyData.n_cells` or `pv.PolyData.n_faces_strict` instead.
+                See the documentation in '`pv.PolyData.n_faces` for more information.""",
+                PyVistaDeprecationWarning,
+            )
```

Comment:
is recommended to make it a function as well as `use_strict_n_faces` in my mind. It is prefer to avoid using global within a class as much as possible.